frontend/src/views/Game.tsx

```diff
@@ -180,26 +192,35 @@ function GamePage() {
 
     // Get the result of promises and set the default code list.
     Promise.all(promises).then((result) => {
-      const codeList = [];
+      const newCodeList = [];
+      const newLanguageList = [];
+      const codeMap = result;
 
       for (let i = 0; i < result.length; i += 1) {
-        codeList.push(result[i][language[i]]);
+        newCodeList.push(result[i][Language.Python]);
+        newLanguageList.push(Language.Python);
       }
 
       // If previous code and language specified, save those as defaults
-      if (code) {
-        codeList[0] = code;
+      for (let i = 0; i < result.length; i += 1) {
+        const temp = getSubmission(i, playerSubmissions);
+
+        if (temp != null) {
+          newCodeList[i] = temp.code;
+          codeMap[i][temp.language as Language] = temp.code;
+          newLanguageList[i] = temp.language as Language;
+          setCurrentProblem(i);
+        }
       }
 
-      // Set this user's current code and language
-      setCurrentCode(codeList);
-      setCurrentLanguage(language);
-
-      setDefaultCodeList(result);
+      // Set this user's current code
+      setCodeList(newCodeList);
+      setLanguageList(newLanguageList);
+      setDefaultCodeList(codeMap);
```

Comment:
Does `setDefaultCodeList` need to be added to the dependencies?

Reply:
I checked, and it seems like it doesn't, but I don't really understand why. So should I remove `setDefaultCodeList` from the dependency list?